src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/comments/ScannerWithCommentEnabledTest.kt

```diff
@@ -0,0 +1,363 @@
+
+        val sut = constructScanner(
+            """
+            abc: |- # Comment
+                def
+                hij
+
+
+            """.trimIndent()
+        )
+
+        assertTokensEqual(
+            expectedList = expected,
+            expectedScalarValueList = expectedScalarValue,
+            sut = sut,
+        )
+    }
+
+    test("block scalar - keep new line - all new lines at and") {
+        val expected = listOf(
+            ID.StreamStart,
+            ID.BlockMappingStart,
+            ID.Key, ID.Scalar, ID.Value, ID.Comment, ID.Scalar,
+            ID.BlockEnd,
+            ID.StreamEnd,
+        )
+        val expectedScalarValue = listOf("abc", "def\nhij\n\n")
+
+        val sut = constructScanner(
+            """
+            abc: |+ # Comment
+                def
+                hij
+
+
+            """.trimIndent()
+        )
+
+        assertTokensEqual(
+            expectedList = expected,
+            expectedScalarValueList = expectedScalarValue,
+            sut = sut,
+        )
+    }
+
+    test("directive line end comment") {
+        val expected = listOf(
+            ID.StreamStart,
+            ID.Directive,
+            ID.Comment,
+            ID.StreamEnd,
+        )
+
+        val sut = constructScanner("%YAML 1.1 #Comment\n")
+
+        assertTokensEqual(
+            expectedList = expected,
+            sut = sut,
+        )
+    }
+})
+
+private fun constructScanner(input: String): Scanner {
+    val settings = LoadSettings.builder().setParseComments(true).build()
+    return ScannerImpl(settings, StreamReader(settings, input))
+}
+
+private fun assertTokensEqual(
+    expectedList: List<ID>,
+    expectedScalarValueList: List<String>? = null,
+    sut: Scanner,
+) {
+    val expectedIterator = expectedList.iterator()
+    val expectedScalarValueIterator = expectedScalarValueList?.iterator()
+
+    while (!sut.checkToken(ID.StreamEnd)) {
+        val token = sut.next()
+        assertTokenEquals(
+            expectedIdIterator = expectedIterator,
+            expectedScalarValueIterator = expectedScalarValueIterator,
+            token = token,
+        )
+    }
+
+    val token = sut.peekToken()
+    assertTokenEquals(
+        expectedIdIterator = expectedIterator,
+        expectedScalarValueIterator = expectedScalarValueIterator,
+        token = token,
+    )
+    expectedIterator.hasNext().shouldBeFalse()
+}
+
+private fun assertTokenEquals(
+    expectedIdIterator: Iterator<ID>,
+    expectedScalarValueIterator: Iterator<String>?,
+    token: Token,
+) {
+    expectedIdIterator.hasNext().shouldBeTrue()
+    val expectedValue = expectedIdIterator.next()
+    val id = token.tokenId
+    println("Expected value: $expectedValue")
+    println("Actual   value: $id")
+    println()
+    expectedValue shouldBeSameInstanceAs id
+
+    if (expectedScalarValueIterator != null && id == ID.Scalar) {
+        expectedScalarValueIterator.next() shouldBe (token as ScalarToken).value
+    }
```

Comment:
suggestion: I think we could replace calling `println` with call to `withClue` method

```kotlin
withClue(
    {
        buildString {
            appendLine("Expected value: $expectedValue")
            appendLine("Actual value: $id")
        }
    }
) {
    expectedValue shouldBeSameInstanceAs id
    if (expectedScalarValueIterator != null && id == ID.Scalar) {
        expectedScalarValueIterator.next() shouldBe (token as ScalarToken).value
    }
}
```

It looks a bit ugly but the main difference would be that this clue will appear in the error message in case of test failure (the `println` results will be visible only in the test's stdout)

Reply:
Oh, this was supposed to be used only in debugging, I forgot to remove it! I think we don't need the clue here, to not obscure the test.